opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -76,14 +79,24 @@ _winfo_create(opal_common_ucx_wpool_t *wpool)
 static void
 _winfo_reset(opal_common_ucx_winfo_t *winfo)
 {
+    if (winfo->inflight_req != UCS_OK) {
+        opal_common_ucx_wait_request(winfo->inflight_req, winfo->worker,
+                                     "opal_common_ucx_flush");
+        winfo->inflight_req = UCS_OK;
+    }
+
+    assert(winfo->global_inflight_ops == 0);
+
     if(winfo->comm_size != 0) {
         size_t i;
         for (i = 0; i < winfo->comm_size; i++) {
             if (NULL != winfo->endpoints[i]){
                 ucp_ep_destroy(winfo->endpoints[i]);
             }
+            assert(winfo->inflight_ops[i] == 0);
```

Comment:
Is this guaranteed externally?

Reply:
Yes, this is guaranteed. At this point, each process should already call flush to guarantee completion, which means counter should be set to 0.